packages/vite/src/executors/build/build.impl.ts

```diff
@@ -54,6 +59,25 @@ export async function* viteBuildExecutor(
       },
       context
     );
+  } else if (options.generatePackageJson) {
```

Comment:
We should merge the detected dependencies into the original `package.json` file. There could be other things like the name or keyword fields that users want to maintain manually.

Reply:
Seems like the `createPackageJson` helper already does this so I've added the test. Also made `generatePackageJson` take precedence as existing `package.json` files were just being copied over